src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -366,12 +387,80 @@ class PremisesController(
         service = body.serviceName.value,
         originalArrivalDate = body.arrivalDate,
         originalDepartureDate = body.departureDate,
-        createdAt = OffsetDateTime.now(),
+        createdAt = bookingCreatedAt,
         application = if (associateWithOnlineApplication) newestSubmittedOnlineApplication else null,
         offlineApplication = if (associateWithOfflineApplication) newestOfflineApplication else null
       )
     )
 
+    if (body is NewApprovedPremisesBooking && booking.application != null) {
```

Comment:
It's no longer possibe to create a CAS1 Booking without an associated Application, there might still be some in dev/preprod that this will guard against.  The main reason this check is here though is that we need an associated Online Application to produce a Domain Event since we need the Delius Event Number which we don't know for Offline Applications.  Have made this a bit clearer in the refactored version: https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/441/files#diff-6c0f9c23780af629762d73597aa46737ce4787a4616d75563fcf92d23039827fR142